### ISSUE TITLE
[FEATURE REQUEST] Add a return evaluation to unless & onlyif requisites 

### ISSUE THREAD

**Issue by Oloremo** [saltstack/salt#57470]
**Is your feature request related to a problem? Please describe.**
Assume this example:

we have a consul execution module: `consul.get` that returns this data on a non-existent key
```
local:
  data: Key not found.
  res: false
```
And state like this:
```
test:
  test.nop:
    - name: foo
    - onlyif:
      - fun: consul.get
        consul_url: http://127.0.0.1:8500
        key:  not-existing

```
will pass
```
          ID: test
    Function: test.nop
        Name: foo
      Result: True
     Comment: Success!
     Started: 10:59:01.517954
    Duration: 3091.398 ms
     Changes:
```

This is due to a very naive truthness check in the current implementation and due to the fact that there is no structure or strict format of the execution modules output:

https://github.com/saltstack/salt/blob/v3000.3/salt/state.py#L915-L923 (onlyif)
https://github.com/saltstack/salt/blob/v3000.3/salt/state.py#L953-L970 (unless)

**Describe the solution you'd like**
I think the easiest solution would be an ability to drill into return data and evaluate it like in the example above I'd check if the `res` is `True`.

**Comment by Akm0d** [comment]
Thank you for the feature request!  FYI This is something we're baking into `idem` early on in that project.  Does [salt-check](https://github.com/wcannon/salt-check) do something similar to what is being requested here for salt?

**Comment by mchugh19** [comment]
Feature addressed in #57504
